source/material_model/dynamic_friction.cc

```diff
@@ -0,0 +1,444 @@
+                                                                   n_fields,
+                                                                   "Specific heats");
+          prm.enter_subsection("Viscosities");
+          {
+            minimum_viscosity  = prm.get_double ("Minimum viscosity");
+            maximum_viscosity  = prm.get_double ("Maximum viscosity");
+            reference_strain_rate    = prm.get_double ("Reference strain rate");
+
+            mu_s = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Coefficients of static friction"))),
+                                                                  n_fields,
+                                                                  "Coefficients of static friction");
+            mu_d = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Coefficients of dynamic friction"))),
+                                                                  n_fields,
+                                                                  "Coefficients of dynamic friction");
+            cohesions = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Cohesions"))),
+                                                                  n_fields,
+                                                                  "Cohesions"); 
+            background_viscosities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Background Viscosities"))),
+                                                                  n_fields,
+                                                                  "Background Viscosities"); 
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::compositional_fields;
+      this->model_dependence.density = NonlinearDependence::temperature | NonlinearDependence::compositional_fields;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::compositional_fields;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::compositional_fields;
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(DynamicFriction,
+                                   "dynamic friction",
+                                   "This model is for use with an arbitrary number of compositional fields, where each field"
+                                   "represents a rock type which can have completely different properties from the others."
```

Comment:
Same here. The reason is that this text is copied into the latex of the manual, and in its current form there will be a space missing.